formats/Gantt/src/Gantt.php

```diff
@@ -0,0 +1,311 @@
+	}
+
+	private function getPriorityMapping() {
+		return $this->mPriorityMapping;
+	}
+
+	private function setStatusMapping( $statusMapping ) {
+		$this->mStatusMapping = $statusMapping;
+	}
+
+	private function getStatusMapping() {
+		return $this->mStatusMapping;
+	}
+
+	private function setAxisFormat( $axisFormat ) {
+		$this->mAxisFormat = $axisFormat;
+	}
+
+	private function getAxisFormat() {
+		return $this->mAxisFormat;
+	}
+
+	private function setTitle( $title ) {
+		$this->mTitle = $title;
+	}
+
+	private function getTitle() {
+		return $this->mTitle;
+	}
+
+	private function getSections() {
+		return $this->mSections;
+	}
+
+	private function getTasks() {
+		return $this->mTasks;
+	}
+
+	/**
+	 * Adds a new Task to array
+	 *
+	 * @param string $taskID
+	 * @param string $taskTitle
+	 * @param array $status
+	 * @param array $priority
+	 * @param string $startDate
+	 * @param string $endDate
+	 *
+	 */
+
+	public function addTask( $taskID, $taskTitle, $status, $priority, $startDate, $endDate ) {
+		$task = new GanttTask();
+		$task->setID( $taskID );
+		$task->setTitle( $taskTitle );
+		$task->setTaskParam( $status, $this->getStatusMapping(), "status" );
+		$task->setTaskParam( $priority, $this->getPriorityMapping(), "priority" );
+		$task->setStartDate( $startDate );
+		$task->setEndDate( $endDate );
+
+		$this->mTasks[$taskID] = $task;
+	}
+
+
+	/**
+	 * Creats a new Section with related tasks
+	 *
+	 * @param string $sectionID
+	 * @param string $sectionTitle
+	 * @param string $startDate
+	 * @param string $endDate
+	 * @param string $taskID
+	 *
+	 */
+	public function addSection( $sectionID, $sectionTitle, $startDate, $endDate, $taskID ) {
+
+		$sections = $this->getSections();
+		if ( count( $sections ) != 0 ) {
+
+			if ( array_key_exists( $sectionID, $sections ) ) {
+
+				foreach ( $sections as $sectionObj ) {
+					if ( $sectionObj->getID() == $sectionID ) {
+						if ( $sectionObj->getEarliestStartDate() > $startDate ) {
+							$sectionObj->setEarliestStartDate( $startDate );
+						}
+						if ( $sectionObj->getLatestEndDate() < $endDate ) {
+							$sectionObj->setLatestEndDate( $endDate );
+						}
+						$sectionObj->addTask( $taskID );
+					}
+				}
+			} else {
+				$this->createNewSection( $sectionID, $sectionTitle, $startDate, $endDate, $taskID );
+			}
+
+		} else {
+			// initialize GanttSection Array
+			$this->createNewSection( $sectionID, $sectionTitle, $startDate, $endDate, $taskID );
+		}
+	}
+
+	/**
+	 * Sort Tasks based on the sortkey
+	 *
+	 * @param GanttTask $a
+	 * @param GanttTask $b
+	 *
+	 * @return integer
+	 */
+	public function sortTasks( $a, $b ) {
+
+		$sortKey = $this->getSortKey();
+
+		// sort based on start/end date or title
+		if ( $sortKey == 'title' ) {
+			// sort based on title
+			return strcmp( $a->getTitle(), $b->getTitle() );
+		} else {
+			if ( strpos( $sortKey, 'date' ) !== false ) {
+				if ( strpos( $sortKey, 'start' ) !== false ) {

```

Comment:
Remove sort functionality, this can be done with inline query standard sort and order property
Line: 126-182

Commit: 6aa6b90db221170166efcb53f983a06dfe394b03